streams/src/test/java/org/apache/kafka/streams/state/internals/OffsetCheckpointTest.java

```diff
@@ -85,6 +85,25 @@ public void shouldNotWriteCheckpointWhenNoOffsets() throws IOException {
         checkpoint.delete();
     }
 
+    @Test
+    public void shouldDeleteExistingCheckpointWhenNoOffsets() throws IOException {
+        final File file = TestUtils.tempFile();
+        final OffsetCheckpoint checkpoint = new OffsetCheckpoint(file);
+
+        final Map<TopicPartition, Long> offsets = Collections.singletonMap(new TopicPartition(topic, 0), 1L);
+
+        checkpoint.write(offsets);
+
+        assertTrue(file.exists());
+        assertEquals(offsets, checkpoint.read());
+
+        checkpoint.write(Collections.emptyMap());
+
+        assertFalse(file.exists());
+
+        assertEquals(Collections.<TopicPartition, Long>emptyMap(), checkpoint.read());
```

Comment:
Suggestion for better readability:

```suggestion
        assertThat(file.exists(), is(true));
        assertThat(offsets, is(checkpoint.read()));

        checkpoint.write(Collections.emptyMap());

        assertThat(file.exists(), is(false));
        assertThat(Collections.<TopicPartition, Long>emptyMap(), is(checkpoint.read()));
```

I verified that the suggested code does not break the test. So, you can directly add the suggesion in the GitHub UI.

Reply:
Thanks for the suggestion. I had to add imports but otherwise looks good 👍

Reply:
Ouch! Sorry about that!